tools/devops/automation/scripts/bash/build-mlaunch-nuget.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -ex
+
+# env var should have been defined by the CI
+if test -z "$XAM_TOP"; then
+    echo "Variable XAM_TOP is missing."
+    exit 1
+fi
+
+if test -z "$MACCORE_TOP"; then
+    echo "Variable MACCORE_TOP is missing."
+    exit 1
+fi
+
+cd "$XAM_TOP"
+
+MACCORE_HASH=$(cd "$MACCORE_TOP" && git log -1 --pretty=%h)
+
+if nuget list -source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json -AllVersions -Prerelease Microsoft.DotNet.Mlaunch | grep $MACCORE_HASH; then
+    echo "Mlaunch revision $MACCORE_HASH is already published as nupkg"
+    exit 0
+fi
+
+# Package mlaunch as .nupkg
+echo "Packaging mlaunch revision $MACCORE_HASH as nupkg..."
+
+DOTNET_NUPKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_NUPKG_DIR | grep "^DOTNET_NUPKG_DIR=" | sed -e 's/^DOTNET_NUPKG_DIR=//')
+MLAUNCH_WORK_DIR="$DOTNET_NUPKG_DIR/mlaunch-staging"
+
+rm -rf "$MLAUNCH_WORK_DIR"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/bin"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+
+DOTNET6=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET6 | grep "^DOTNET6=" | sed -e 's/^DOTNET6=//')
+IOS_DESTDIR=$(make -C tools/devops print-abspath-variable VARIABLE=IOS_DESTDIR | grep "^IOS_DESTDIR=" | sed -e 's/^IOS_DESTDIR=//')
+MONOTOUCH_PREFIX=$(make -C tools/devops print-abspath-variable VARIABLE=MONOTOUCH_PREFIX | grep "^MONOTOUCH_PREFIX=" | sed -e 's/^MONOTOUCH_PREFIX=//')
+XCODE_VERSION=$(grep XCODE_VERSION= "$XAM_TOP/Make.config" | sed 's/.*=//')
+
+# Copy mlaunch to staging area
+cp -c -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+cp -c "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
+
+# Add the .csproj we will use to create the .nupkg
+cp -c "$XAM_TOP/tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj" "$MLAUNCH_WORK_DIR"
+
+# We need to override global.json to use .NET 6.0
+cp -c "$XAM_TOP/global6.json" "$MLAUNCH_WORK_DIR/global.json"
+
+# We have to build from within the dir to respect the global.json
+cd "$MLAUNCH_WORK_DIR"
+"$DOTNET6" pack --version-suffix "$MACCORE_HASH" /p:VersionPrefix=$XCODE_VERSION
```

Comment:
Sadly we might need more than one mlaunch release for each version of Xcode, bug happens :(

> The current package name is Microsoft.DotNet.Mlaunch.12.5.0-cddbd1915d.nupkg

If you do an update (for Xcode 12.5.0) then only the digest (cddbd1915d) will change. 
It will still be a unique name but no one will be able to figure out which is the newest version.

Something more (like the commit distance or however net6 compute `21276.13`) is needed to distinguish the published packages.

Reply:
I searched Arcade and we use this formula in .NET 6:
```
(yy * 1000 + mm * 50 + dd).(BuildRevision)
```

So today's number is:
```
21 * 1000 + 06 * 50 + 10 = 21310.x
```

source: https://github.com/dotnet/arcade/blob/e81d8c9bdc1de22623657afb23fab210dd04ca82/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L57

So I can do `12.5.0-21310.1-cddbd1915d`?
Or we rather want `21310.1-12.5.0-cddbd1915d`?

Reply:
Or I have seen in Xamarin previously `+cddbd1915d` or `+sha_cddbd1915d` I think

Reply:
I think something like this could work:

> 12.5.0-21310.1.cddbd1915d

Reply:
So, we now have `Microsoft.DotNet.Mlaunch.12.5.0-21310.9.cddbd1915d.nupkg`

Built here:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4857230&view=logs&j=600d49c6-e96b-5e8b-2c3a-9f351e2e5848&t=c9117ceb-ee46-564f-0d31-c57e59243b74&l=181